.github/workflows/selfdrive_tests.yaml

```diff
@@ -25,7 +25,7 @@ env:
   DOCKER_LOGIN: docker login ghcr.io -u ${{ github.actor }} -p ${{ secrets.GITHUB_TOKEN }}
   BUILD: selfdrive/test/docker_build.sh base
 
-  RUN: docker run --shm-size 1G -v $PWD:/tmp/openpilot -w /tmp/openpilot -e CI=1 -e PRE_COMMIT_HOME=/tmp/pre-commit -e PYTHONWARNINGS=error -e FILEREADER_CACHE=1 -e PYTHONPATH=/tmp/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v $GITHUB_WORKSPACE/.ci_cache/pre-commit:/tmp/pre-commit -v $GITHUB_WORKSPACE/.ci_cache/scons_cache:/tmp/scons_cache -v $GITHUB_WORKSPACE/.ci_cache/comma_download_cache:/tmp/comma_download_cache -v $GITHUB_WORKSPACE/.ci_cache/openpilot_cache:/tmp/openpilot_cache $BASE_IMAGE /bin/bash -c
+  RUN: docker run --shm-size 2G -v $PWD:/tmp/openpilot -w /tmp/openpilot -e CI=1 -e PRE_COMMIT_HOME=/tmp/pre-commit -e PYTHONWARNINGS=error -e FILEREADER_CACHE=1 -e PYTHONPATH=/tmp/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v $GITHUB_WORKSPACE/.ci_cache/pre-commit:/tmp/pre-commit -v $GITHUB_WORKSPACE/.ci_cache/scons_cache:/tmp/scons_cache -v $GITHUB_WORKSPACE/.ci_cache/comma_download_cache:/tmp/comma_download_cache -v $GITHUB_WORKSPACE/.ci_cache/openpilot_cache:/tmp/openpilot_cache $BASE_IMAGE /bin/bash -c
```

Comment:
Any way we can check the runner logs to see if it ran out of memory or not? It does seem probable, wonder what has increased in memory usage for the last few months though. From adding more tests to this one job, or a memory leak?

Reply:
test_car_interfacees.py seems to increase my machine's memory usage by 1GB with 8 jobs!

Reply:
`pytest-memray`, not sure how accurate this is:

```
Allocation results for selfdrive/car/tests/test_car_interfaces.py::TestCarInterfaces::test_car_interfaces_007_BUICK_REGAL at the high watermark

         📦 Total memory allocated: 39.1MiB
         📏 Total allocations: 390
         📊 Histogram of allocation sizes: | ▄█  |
         🥇 Biggest allocating functions:
                - get_kalman_gain:/home/batman/openpilot/common/simple_kalman.py:7 -> 32.0MiB
                - getstate:/home/batman/.pyenv/versions/3.11.4/lib/python3.11/random.py:168 -> 1.0MiB
                - <dictcomp>:/home/batman/openpilot/selfdrive/test/fuzzy_generation.py:50 -> 1.0MiB
                - __init__:/home/batman/openpilot/selfdrive/car/gm/carstate.py:18 -> 1.0MiB
                - getstate:/home/batman/.pyenv/versions/3.11.4/lib/python3.11/random.py:168 -> 1.0MiB
```

Reply:
after a bit of testing and research, I'm not sure if this is an issue of memory usage. 

https://github.com/pytest-dev/pytest/issues/3216#issuecomment-742143197. After serializing some tests, everything passes pretty consistently

Reply:
You can monitor the usage for a run to see how close it is to 1GB currently, that'll give you a good idea if it contributes.

Reply:
other pytest monitoring tools seems to only measure singular test only which makes it hard to measure memory usage. I manually look at `htop` and see that running a `pytest -n8 'no slow'` takes > 2.5GB